activejob/lib/active_job/test_helper.rb

```diff
@@ -7,8 +8,14 @@ module TestHelper
 
     included do
       def before_setup
-        @old_queue_adapter = queue_adapter
-        ActiveJob::Base.queue_adapter = :test
+        test_adapter = ActiveJob::QueueAdapters::TestAdapter.new
+
+        @old_queue_adapters = (ActiveJob::Base.subclasses << ActiveJob::Base).map do |klass|
+          [klass, klass.queue_adapter].tap do
+            klass.queue_adapter = test_adapter
+          end
+        end
```

Comment:
Since class attributes no longer inherit from superclass once they're written, assigning `#queue_adapter` on classes the didn't have it explicitly set will leak side effects between test cases (since changing a superclass attr won't be inherited by its subclass afterward)